user-guide/modules/ROOT/pages/build-options.adoc

```diff
@@ -0,0 +1,152 @@
+= Build Options
+:navtitle: Build Options
+
+This section covers a number of build scenarios that you may come across if your project has specific, legacy, or non-standard, components or processes.
+
+== Build Individual Libraries
+
+If you only need specific libraries and want to avoid building the entire Boost library suite, which can be time-consuming, then use the `--with-` parameter with the `b2` command to specify the individual libraries that you want to build. For example, to build the boost:thread[] library:
+
+[source,bash]
+----
+b2 --with-thread
+----
+
+Note:: Some libraries are header-only, meaning they do not need to be built before use. Examples include boost:asio[], boost:bind[], boost:function[], and boost:lambda[]. You don't need to use `--with-` for these libraries since they're included directly in your source code and don't need to be pre-compiled.
+
+To build multiple libraries, say boost:thread[] and boost:filesystem[], simply repeat the parameter:
+
+[source,bash]
+----
+b2 --with-thread --with-filesystem --with-system
+----
+
+The reason why we need `--with-system` is that it is a dependency of boost:filesystem[]. 
```

Comment:
@alandefreitas I did not get this section right. If I build a boost library that can be built (ie: not header-only, such as filesystem) then indeed the dependencies are built too. The problem arises when I try to build a boost library like asio (which is header-only) BUT it has dependencies on libraries that need to be built (system etc.).  In this case the b2 command simply errors out - I have some mail out trying to verify how this should be handled.

Reply:
> The problem arises when I try to build a boost library like asio (which is header-only) BUT it has dependencies on libraries that need to be built

Yes. That makes more sense. I think the way `b2` looks at it (because it sucks in that regard) is that you can't build asio because it's not a "real" target in the `b2` sense because there's nothing to build so there can't be a `--with-asio` and then it can't find its dependencies because it doesn't exist. The way `cmake` handles it is much simpler because it sees `boost::asio` as an interface target that just propagates its properties, including any flags, include paths and dependencies.

If it's all about header-only libraries only, then the section looks OK as long as we adapt the example to a built library then to a header-only library to show this as a limitation. See the other point I also made about the `cmake` section though.